python/common/business-registry-model/pyproject.toml

```diff
@@ -33,10 +33,168 @@ freezegun = "^1.5.1"
 pytest = "^8.3.5"
 zimports = "^0.6.1"
 black = "^25.1.0"
+pytest-cov = "^6.0.0"
 
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".vscode",
+    "dist",
+    "htmlcov",
+    "venv",
+    "migrations",
+    "devops",
+    ".history"
+]
 
 [tool.poetry.dependencies]
 sql-versioning = {develop = true}
 [build-system]
 requires = ["poetry-core>=2.0.0,<3.0.0"]
 build-backend = "poetry.core.masonry.api"
+
+
+# Same as Black.
+line-length = 120
+indent-width = 4
+
+# Assume Python 3.12
+#target-version = "py312"
+
+[tool.ruff.format]
+# Like Black, use double quotes for strings.
+quote-style = "double"
+
+# Like Black, indent with spaces, rather than tabs.
+indent-style = "space"
+
+# Like Black, respect magic trailing commas.
+skip-magic-trailing-comma = false
+
+# Like Black, automatically detect the appropriate line ending.
+line-ending = "auto"
+
+# Enable auto-formatting of code examples in docstrings. Markdown,
+# reStructuredText code/literal blocks and doctests are all supported.
+#
+# This is currently disabled by default, but it is planned for this
+# to be opt-out in the future.
+docstring-code-format = true
+
+# Set the line length limit used when formatting code snippets in
+# docstrings.
+#
+# This only has an effect when the `docstring-code-format` setting is
+# enabled.
+docstring-code-line-length = "dynamic"
+
+[tool.ruff.lint]
+select = [
+    "B",   # flake8-bugbear
+    "C4",  # flake8-comprehensions
+    "ERA", # flake8-eradicate/eradicate
+    "I",   # isort
+    "N",   # pep8-naming
+    "PIE", # flake8-pie
+    "PL",  # pyLint
+    "PGH", # pygrep
+    "RUF", # ruff checks
+    "SIM", # flake8-simplify
+    "T20", # flake8-print
+    "TCH", # flake8-type-checking
+    "TID", # flake8-tidy-imports
+    "UP",  # pyupgrade
+]
+#ignore = [
+#    "B904", # use 'raise ... from err'
+#    "B905", # use explicit 'strict=' parameter with 'zip()'
+#    "N999"
+#]
+[tool.ruff.lint.per-file-ignores]
+"**/tests/*" = [
+    "B",
+    "C",
+    "ERA",
+    "N",
+    "PLR",
+    "RUF",
+    "SIM",
+    "T2",
+    ]
+"**/business_model_migrations/*" = [
+    "ERA",
+    "PLR",
+    "T",
+    ]
+#"src/business_model/utils/datetime.py" = [
+#    "N801",
+#    ]
+"**/business_model/*" = [
+    "N801",
+    "RUF012",
+    "RUF013",
+    "SIM",
+    "TC001",
+    ]
+
+
+# Allow fix for all enabled rules (when `--fix`) is provided.
+fixable = ["ALL"]
+unfixable = [
+    "ERA", # do not autoremove commented out code
+]
+
+# Allow unused variables when underscore-prefixed.
+#dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
+
+[tool.ruff.lint.isort]
+known-first-party = ["business_model"]
```

Comment:
Do we want to add 'sql_versioning_alt' and 'registry_schemas' ? It helps to know if its our own stuff when we get dependency issues (mainly for newer devs who won't be familiar with the common libraries)